pkg/rancher-desktop/pages/volumes/files/editor.vue

```diff
@@ -0,0 +1,400 @@
+        throw new Error('Invalid file path format');
+      }
+      return path;
+    },
+    hasNamespaceSelected() {
+      return this.settings?.containerEngine?.name === ContainerEngine.CONTAINERD && this.settings?.containers?.namespace;
+    },
+    isModified() {
+      return this.content !== this.originalContent;
+    },
+  },
+  mounted() {
+    this.$store.dispatch('page/setHeader', {
+      title: this.t('volumes.editor.title'),
+      description: this.filePath.split('/').pop(),
+    });
+
+    ipcRenderer.on('settings-read', this.onSettingsRead);
+    ipcRenderer.send('settings-read');
+
+    window.addEventListener('beforeunload', this.handleBeforeUnload);
+  },
+  beforeDestroy() {
+    ipcRenderer.off('settings-read', this.onSettingsRead);
+    window.removeEventListener('beforeunload', this.handleBeforeUnload);
+    if (this.saveMessageTimer) {
+      clearTimeout(this.saveMessageTimer);
+    }
+  },
+  methods: {
+    async onSettingsRead(event, settings) {
+      this.settings = settings;
+      await this.loadFile();
+    },
+    async loadFile() {
+      if (window.ddClient && this.isK8sReady && this.settings) {
+        this.ddClient = window.ddClient;
+        await this.readFileContent();
+      }
+    },
+    async readFileContent() {
+      try {
+        this.error = null;
+
+        const containerFilePath = `/volume${this.filePath}`;
+        const catCommand = [
+          'run', '--rm',
+          '-v', `${this.volumeName}:/volume:ro`,
+          'busybox',
+          'cat', containerFilePath
+        ];
+
+        const execOptions = {cwd: '/'};
+        if (this.hasNamespaceSelected) {
+          execOptions.namespace = this.hasNamespaceSelected;
+        }
+
+        const {stderr, stdout} = await this.ddClient.docker.cli.exec(
+          catCommand[0],
+          catCommand.slice(1),
+          execOptions
+        );
+
+        if (stderr && !stderr.includes('level=warning')) {
+          throw new Error(stderr);
+        }
+
+        this.content = stdout;
+        this.originalContent = stdout;
+        this.fileSize = new TextEncoder().encode(stdout).length;
+        this.isLoading = false;
+      } catch (error) {
+        console.error('Error reading file:', error);
+        this.error = this.t('volumes.editor.readError', {error: error.message});
+        this.isLoading = false;
+      }
+    },
+    async saveFile() {
+      if (!this.isModified || this.isSaving) return;
+
+      try {
+        this.isSaving = true;
+        this.error = null;
+
+        const containerFilePath = `/volume${this.filePath}`;
+
+        // Create a temporary file with the new content
+        const tempFileName = `temp_${Date.now()}.txt`;
+        const echoCommand = [
+          'run', '--rm',
+          '-v', `${this.volumeName}:/volume`,
+          'busybox',
+          'sh', '-c',
+          `cat > /volume/${tempFileName} << 'EOF'
+${this.content}
+EOF
+mv /volume/${tempFileName} ${containerFilePath}`
```

Comment:
This means we change the file permissions, right?

Also, this would have… interesting effects if the file contains `EOF` on a line by itself (e.g. somebody writing a shell script using heredoc)… Can we just feed it in via stdin instead? Or base64 it and then decode it in the container…